events/tests/test_utils.py

```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+
+import pytest
+import pytz
+from django.conf import settings
+from django.utils import timezone
+from freezegun import freeze_time
+from rest_framework.exceptions import ParseError
+
+from events.utils import parse_time
+
+
+@pytest.fixture
+def local_tz():
+    return pytz.timezone(settings.TIME_ZONE)
+
+
+@pytest.mark.django_db
+def test_parse_time_iso8601(local_tz):
+    assert parse_time("2022-07-17T17:17:17+03:00", True) == (
+        local_tz.localize(datetime(2022, 7, 17, 17, 17, 17)),
+        True,
+    )
+
+
+@pytest.mark.django_db
+def test_parse_time_iso8601_invalid_offset(local_tz):
+    with pytest.raises(ParseError) as ex:
+        parse_time("2022-07-17T17:17:17+25:00", True)
+        assert "out of bounds" in str(ex.value)
```

Comment:
Afaik, `parse_time` raises so the assertation clause is never reached unless you make the following change. And making that change seems to make the test fail since the error message is not what is expected here. :eyes: 
```suggestion
    with pytest.raises(ParseError) as ex:
        parse_time("2022-07-17T17:17:17+25:00", True)
    assert "out of bounds" in str(ex.value)
```

Reply:
Excellent, thanks. Not sure how that happened. Probably copilot :trollface: